src/kvirc/ui/KviWebPackageManagementDialog.cpp

```diff
@@ -60,125 +128,162 @@ KviWebPackageManagementDialog::KviWebPackageManagementDialog(QWidget * pParent)
-
-	m_pToolBar = new QToolBar(pStatus);
-	m_pProgressBar = new QProgressBar(pStatus);
-	m_pProgressBar->setMaximumWidth(220);
-	m_pProgressBar->setMinimumWidth(220);
-	hbox->addWidget(m_pToolBar);
-	hbox->addWidget(m_pProgressBar);
-	hbox->setAlignment(m_pProgressBar, Qt::AlignRight);
-
-	m_pWebView = new QWebEngineView(this);
-	// we managing the links
-	m_pWebView->setPage(new KviWebPackagePage());
-
-	m_pLayout->addWidget(m_pWebView);
-	m_pLayout->setStretchFactor(m_pWebView, 10); // must take most space
-	// disable context menu
-	m_pWebView->setContextMenuPolicy(Qt::NoContextMenu);
-	m_pToolBar->addAction(m_pWebView->pageAction(QWebEnginePage::Back));
-	m_pToolBar->addAction(m_pWebView->pageAction(QWebEnginePage::Forward));
-
-	//connect(pWebView,SIGNAL(loadStarted()),this,SLOT(slotLoadStarted()));
-	connect(m_pWebView, SIGNAL(loadFinished(bool)), this, SLOT(slotLoadFinished(bool)));
-	connect(m_pWebView, SIGNAL(loadProgress(int)), this, SLOT(slotLoadProgress(int)));
-	connect(m_pWebView->page(), SIGNAL(linkClicked(const QUrl &)), this, SLOT(slotLinkClicked(const QUrl &)));
+	m_pContextPopup = new QMenu(this);
+
+	QVBoxLayout * pVBox = new QVBoxLayout(this);
+	KviTalHBox * pHBox = new KviTalHBox(this);
+	pHBox->setContentsMargins(1, 1, 1, 1);
+	pHBox->setSpacing(1);
+	pVBox->addWidget(pHBox);
+
+	m_pPreviewButton = new QToolButton(pHBox);
+	m_pPreviewButton->setIcon(*(g_pIconManager->getBigIcon(KVI_BIGICON_THEME)));
+	m_pPreviewButton->setIconSize(QSize(32, 32));
+	m_pPreviewButton->setToolTip(__tr2qs("Show item preview"));
+	connect(m_pPreviewButton, SIGNAL(clicked()), this, SLOT(showItemPreview()));
+
+	m_pDeleteButton = new QToolButton(pHBox);
+	m_pDeleteButton->setIcon(*(g_pIconManager->getBigIcon(KVI_BIGICON_PACK)));
+	m_pDeleteButton->setIconSize(QSize(32, 32));
+	m_pDeleteButton->setToolTip(__tr2qs("Download selected item"));
+	connect(m_pDeleteButton, SIGNAL(clicked()), this, SLOT(downloadItem()));
+
+	QWidget * w = new QWidget(pHBox);
+	w->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Minimum);
+
+	m_pProgressBar = new QProgressBar(this);
+	pVBox->addWidget(m_pProgressBar);
+
+	m_pListWidget = new KviTalListWidget(this);
+	m_pListWidget->setContextMenuPolicy(Qt::CustomContextMenu);
+	m_pItemDelegate = new KviTalIconAndRichTextItemDelegate(m_pListWidget);
+	m_pItemDelegate->setDefaultIcon(g_pIconManager->getBigIcon(QString(KVI_BIGICON_THEME))->scaled(64, 64, Qt::KeepAspectRatio));
+	m_pItemDelegate->setMinimumSize(QSize(64, 64));
+	m_pItemDelegate->setIconSize(QSize(64, 64));
+	m_pListWidget->setItemDelegate(m_pItemDelegate);
+	m_pListWidget->setMinimumHeight(400);
+	m_pListWidget->setMinimumWidth(520);
+
+	m_pListWidget->setSelectionMode(QAbstractItemView::SingleSelection);
+	m_pListWidget->setSortingEnabled(true);
+	//connect(m_pListWidget, SIGNAL(itemDoubleClicked(QListWidgetItem *)), this, SLOT(applyTheme(QListWidgetItem *)));
+
+	connect(m_pListWidget, SIGNAL(customContextMenuRequested(const QPoint &)),
```

Comment:
Above you used &pointer to function to connect to signal?

Reply:
Sorry, I don't get the question.
The signature of the signal is https://doc.qt.io/qt-6/qwidget.html#customContextMenuRequested
It's a const reference (&) to a QPoint object

Reply:
I meant using `SIGNAL(bar())` vs using `&Foo::bar`

Reply:
The different connect() style is needed to be able to use a lambda function a the slot